loki/transformations/extract/marked.py

```diff
@@ -0,0 +1,132 @@
+# (C) Copyright 2018- ECMWF.
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+
+from loki.analyse import dataflow_analysis_attached
+from loki.expression import Variable
+from loki.ir import (
+    CallStatement, Import, PragmaRegion, Section, FindNodes,
+    FindVariables, MaskedTransformer, Transformer, is_loki_pragma,
+    get_pragma_parameters, pragma_regions_attached
+)
+from loki.logging import info
+from loki.subroutine import Subroutine
+from loki.tools import as_tuple, CaseInsensitiveDict
+
+
+__all__ = ['extract_marked_subroutines']
+
+
+def extract_marked_subroutines(routine):
```

Comment:
I find the naming a bit counterintuitive. 

1. "extract_marked_subroutines" implies to me that subroutines with some kind of marker are extracted from somewhere. Wouldn't something that makes it clear that this acts on pragma regions be more appropriated?
2. Any particular reason for calling this "extract" instead of outlining, which seems to be the more common term for this in compiler theory? Would also more directly highlight the complementary nature to inlining. The module could still be called "extract".

Reply:
Yes, agreed. I simply mimicked what was there, but calling it "outlining" is indeed clearer. So, we would have `loki.transformations.extract.internals` and `loki.transformations.extract.outline`, where the latter provides `outline_pragma_regions`?

Reply:
Should we rename the pragmas then too? `!$loki outline` / `!$loki end outline`?

Reply:
Yes, your suggested module naming sounds good to me.
I hadn't thought about the pragmas, but your suggestion does indeed make sense. Reading that in a Fortran source file also makes (imho) immediately clear what is going to happen there.